### PR TITLE
Usage cli report

### PR DESCRIPTION
**Pull request by cesnietor**
Cli command that outputs as stdout a table with the storage group tenant's bucket's average usage per defined period of time:
```
$ ./m3 cluster sc sg usage --storage-cluster=my-dc-rack-1 --storage-group=group-1 --help                                                     
NAME:
  m3 cluster storage-cluster storage-group usage - Get storage group average usage per tenant per period 

USAGE:
  m3 cluster storage-cluster storage-group usage [command options] [arguments...]

FLAGS:
  --storage-cluster value  name of the storage cluster
  --storage-group value    name of the storage group
  --from-date value        start date for the report (YYYY-MM-DD)
  --to-date value          end date (inclusive) for the report (YYYY-MM-DD)
  --export-to-csv          exports the report to a csv file
  --help, -h               show help
```
```
./m3 cluster sc sg usage --storage-cluster=my-dc-rack-1 --storage-group=group-1 --from-date=2020-01-01 --to-date=2020-02-28 
|     PERIOD    | ACCOUNT HOLDER |       TENANT        | CUSTOMER | BUCKET | REGION | AVG USAGE [TB] |
|---------------|----------------|---------------------|----------|--------|--------|----------------|
| 2020-January  |                | Agricultural Bank   |          | buck4  |        |       4.668482 |
| 2020-January  |                | Agricultural Bank   |          | buck1  |        |       0.001940 |
| 2020-January  |                | Agricultural Bank   |          | buck2  |        |       0.000000 |
| 2020-January  |                | Agricultural Bank   |          | buck3  |        |       0.000000 |
| 2020-February |                | Agricultural Bank   |          | buck4  |        |       0.023738 |
| 2020-February |                | Agricultural Bank   |          | buck1  |        |       0.001940 |
| 2020-February |                | Agricultural Bank   |          | buck2  |        |       0.000000 |
| 2020-February |                | Agricultural Bank   |          | buck3  |        |       0.000000 |
| 2020-January  |                | Some Random Company |          | buck4  |        |       4.668482 |
| 2020-January  |                | Some Random Company |          | buck1  |        |       0.001940 |
| 2020-January  |                | Some Random Company |          | buck2  |        |       0.000000 |
| 2020-January  |                | Some Random Company |          | buck3  |        |       0.000000 |
| 2020-February |                | Some Random Company |          | buck4  |        |       0.023738 |
| 2020-February |                | Some Random Company |          | buck1  |        |       0.001940 |
| 2020-February |                | Some Random Company |          | buck2  |        |       0.000000 |
| 2020-February |                | Some Random Company |          | buck3  |        |       0.000000 |
````
Also exports the file locally if needed as csv with the flag --export-to-csv
```
./m3 cluster sc sg usage --storage-cluster=my-dc-rack-1 --storage-group=group-1 --from-date=2020-01-01 --to-date=2020-02-28 --export-to-csv
File created: ./cluster-usage-report-2020-01-01-to-2020-02-28.csv
```

### CODE REVIEW

File: cluster/buckets.go [hunk 1]
```diff
@@ -544,6 +545,131 @@ func GetDailyAvgBucketUsageFromDB(ctx *Context, date time.Time) ([]*BucketMetric
...
+			FROM(
+				SELECT 
+					 EXTRACT (YEAR FROM a.last_update) as year,
+					 EXTRACT (MONTH FROM a.last_update) as month,
+					 EXTRACT (DAY FROM a.last_update) as day,
+					 bsize.key as bucket,
+					 bsize.value::FLOAT as usage
+				FROM bucket_metrics as a
+				JOIN json_each_text(a.buckets_sizes) bsize on true
+				WHERE a.last_update >= $1 AND a.last_update <= $2
```
Comment (dvaldivia) [inline cluster/buckets.go]: This field is not indexed, this query is going to be extremely slow over time

File: cluster/buckets.go [hunk 2]
```diff
@@ -544,6 +545,131 @@ func GetDailyAvgBucketUsageFromDB(ctx *Context, date time.Time) ([]*BucketMetric
...
+			avg(usage) as daily_avg_usage
+			FROM(
+				SELECT 
+					 EXTRACT (YEAR FROM a.last_update) as year,
+					 EXTRACT (MONTH FROM a.last_update) as month,
+					 EXTRACT (DAY FROM a.last_update) as day,
+					 bsize.key as bucket,
+					 bsize.value::FLOAT as usage
+				FROM bucket_metrics as a
+				JOIN json_each_text(a.buckets_sizes) bsize on true
```
Comment (dvaldivia) [inline cluster/buckets.go]: I think we should normalize bucket_sizes to avoid this expensive operation

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: cluster/buckets.go (hunk 1 above)
Comment (cesnietor) [reply cluster/buckets.go]: Done

File: cluster/buckets.go (hunk 2 above)
Comment (cesnietor) [reply cluster/buckets.go]: done by adding two columns `bucket_name` and `bucket_size` also this impacted the periodic function that updates the table bucket_metrics (`buckets.go/getTenantMetrics`) and was updated.

File: cluster/storage-cluster.go [hunk 3]
```diff
@@ -151,15 +151,47 @@ func GetStorageGroupByID(ctx *Context, id *uuid.UUID) (*StorageGroup, error) {
...
+
+	err := GetStorageGroupDetails(ctx, &sg)
+	if err != nil {
+		return nil, err
+	}
+	return &sg, nil
+}
+
+// GetStorageGroupByNameNStorageCLuster returns a storage group by name for a particular storage group
+func GetStorageGroupByNameNStorageCLuster(ctx *Context, name string, storageCluster *StorageCluster) (*StorageGroup, error) {
```
Comment (dvaldivia) [inline cluster/storage-cluster.go]: C `Luster` lol

Comment (cesnietor) [reply cluster/storage-cluster.go]: done

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**